### PR TITLE
Cherry-pick #8424 to 6.x: Reduce casting between []byte and string in CRI log processing

### PR DESCRIPTION
**Pull request by exekias**
Cherry-pick of PR #8424 to 6.x branch. Original message: 

* Reduce casting from []byte to string and []byte again.
* If criflags are set to true, default to CRI log processing.
* Refactor CRI and json-file detection.

### CODE REVIEW

**Comment by exekias** [comment]
Do not merge yet, I want to push https://github.com/elastic/beats/pull/8717 here before

**Comment by exekias** [comment]
Should be ready now (once tests pass)

**Comment by exekias** [comment]
This won't make it to 6.5, leaving the PR open for 6.6

**Comment by exekias** [comment]
This can now be merged for 6.6

File: filebeat/reader/docker_json/docker_json.go [hunk 1]
```diff
@@ -186,3 +203,16 @@ func (p *Reader) Next() (reader.Message, error) {
 		return message, err
 	}
 }
+
+func stripNewLine(msg *reader.Message) {
```
Comment (ruflin) [inline filebeat/reader/docker_json/docker_json.go]: I'm pretty sure we have very similar code in Filebeat for stripping new lines in the new line reader.

**Pull request merged**